Update build script for motor control and refactor code for Lint styl…

Seanmullan opened the pull request:
Fixed file structure to include motor control, updated build script and refactored code for Lint style checks.

Seanmullan committed:
Update build script for motor control and refactor code for Lint style checks

mkristien committed:
Add compilation flags to Makefile
Update README with how to lint ignore lines
Add resultant binary to .gitignore

mkristien committed:
Git Hooks work

mkristien committed:
Add script for deploying git hooks

mkristien committed:
Delete temp file

brano2 reviewed (dismissed):
Sorry for the confusion here! When style guide and linter disagree, always follow style guide. I will try to fix the linter soon but it is pretty much impossible to have a perfect linter for C++ that would never raise any false alarms.
In particular, the opening `{` in a function definition really does belong on a separate line. You had it right before.
You can add a `// NOLINT` comments as described [here](https://github.com/Hyp-ed/hyped-2018/blob/mgt-refactor/BeagleBone_black/README.md#new-project-files)

Good catch with the order of includes though! I missed that last time.

brano2 commented:
Or you can wait till the linter is fixed. Should hopefully be today.

Seanmullan commented:
Thanks for clarifying. I will wait until linter is fixed rather than adding //NOLINT to several lines and then having to remove them.

brano2 committed:
Split cleaning and reorder compilation and linting

There is now a separate target to allow for `make clean`ing of linter's
stuff only so that you can re-lint without recompiling.
Also, `make` now first compiles the project and then lints. It is more
important to have code that compiles than code that lints (although the
latter is still super important).

brano2 committed:
Fix opening braces on new line in function definitions

mkristien commented:
please, merge with the latest develop to get the new lint fixes

mkristien committed:
Add lint rule to prevent wrong {} syntax

mkristien committed:
Clean test code in main.cpp

Seanmullan committed:
Refactor motor controller class

Seanmullan committed:
Refactor code with updated linter

Seanmullan committed:
Remove trailing whitespace in main_motor.cpp
